### PR TITLE
Encode filepath to avoid invalid uri errors

### PR DESCRIPTION
**Pull request by aaron-collier**
## Why was this change made? 🤔

Fixes https://github.com/sul-dlss/happy-heron/issues/3237

I have been able to verify that this resolves the bug locally via the console. However, I'm a little unclear on how this may be tested (or if it should). I have a small test (not committed yet) that adds the filename that needs to be encoded, but I believe the test in place is mocking the calls to where we're avoiding the URI core calls that actually raise the error.

I'm going to leave this as-is but welcome pairing on testing, either in this PR or a follow up (or if we decide it's unnecessary).

## How was this change tested? 🤨

⚡ ⚠ If this change has cross service impact, including data writes to shared file systems, ***run [integration tests](https://github.com/sul-dlss/infrastructure-integration-test)*** and/or test in [stage|qa] environment, in addition to specs. ⚡

### CODE REVIEW

**Review by jmartin-sul — APPROVED** [review]
responding to both the issue description and the slack thread about this...

i'm 👍 on this change.  i haven't looked as thoroughly as i possibly could, but i don't see any concerning possible instances of double-escaping.  the `CGI.escape` call in this PR seems to be the only `CGI.escape` call in this codebase itself.  H2 appears to be this codebase's only consumer, and its only `CGI.escape` call is part of building response headers when passing files back from preservation.

it also seems totally appropriate for a helpful client gem to deal with things like escaping parameter values in the requests it builds, esp since the request-building aspect of this method is helpfully boxed away from the caller.  sorta feels like it'd mix levels of abstraction in a confusing way to force the caller to do the escaping? 🤷 

re: tests, even if the tests don't actually result in any of the invocation of the faraday request machinery that's throwing errors, we could maybe spy on the `connection.get` call, and see that the `filepath` value passed to `ls_path` is escaped as expected (e.g. spaces get percent encoded or similar)?  that would feel sufficient to me.  but i'd also be totally comfortable shipping this without the additional test.  either way, would be curious to see the test you wrote but didn't commit.

re: an idea bounced around in the slack thread, i'd be a little hesitant to only escape when a URI error is caught.  feels like it's safer for callers to expect consistent deterministic escaping behavior.  imho, the caller should have a single clear expectation about whether to provide an escaped value or not (i'd vote for not).

codeclimate indicates that this line is covered by the tests, but i can't actually find where, so it's a little hard to figure out what's getting mocked underneath the `connection.get` call.

plus, as you said, you tested manually and found this fixed things.

TLDR consistently applying this sort of param escaping when building a URL seems like the right thing to do to me.  no strong pref on whether a test is added.

feel free to merge if you want, or to add the test you wrote if you want to do that first.

**Review by ndushay — COMMENTED** [review]
This looks good.  You might be able to test if the (mock/stub) connection gets an escaped filepath, which would be good enough in my opinion.

**Comment by aaron-collier** [comment]
@ndushay I actually don't think this will be useful test because what we've found is that we'll just be testing the string we pass, not the encoding.
